src/server/multi_svr.c

```diff
@@ -295,14 +518,367 @@ gen_svr_inst_id(void)
 	unsigned int svr_inst_port;
 	char *svr_inst_id = NULL;
 
-
 	if (gethostname(svr_inst_name, PBS_MAXHOSTNAME) == 0)
-        	get_fullhostname(svr_inst_name, svr_inst_name, PBS_MAXHOSTNAME);
+		get_fullhostname(svr_inst_name, svr_inst_name, PBS_MAXHOSTNAME);
 
 	svr_inst_port = pbs_conf.batch_service_port;
 
 	pbs_asprintf(&svr_inst_id, "%s:%d", svr_inst_name, svr_inst_port);
 
 	return svr_inst_id;
+}
 
+/**
+ * @brief find the number of peer server replies
+ * which needs to be ack'd
+ * 
+ * @return int 
+ */
+int
+num_pending_peersvr_rply(void)
+{
+	server_t *psvr;
+	int ct = 0;
+	svrinfo_t *svr_info;
+
+	for (psvr = GET_NEXT(peersvrl);
+	     psvr; psvr = GET_NEXT(psvr->mi_link)) {
+		svr_info = psvr->mi_data;
+		ct += svr_info->pending_replies;
+	}
+
+	return ct;
+}
+
+/**
+ * @brief Go through peer server list and
+ * poke if any of them is down.
+ * 
+ * If the application layer came to know about the network failure,
+ * it will close the peer server connection.
+ * The server will periodically attempt to re-connect and
+ * the initial exchange will get triggered upon reconnecting.
+ * 
+ * poke is one such re-connecting mechanism.
+ * Peer-server will also attempts to reconnect when it has some data
+ * to be sent to peer server in the form of resource usage / commands.
+ */
+void
+poke_peersvr(void)
+{
+	server_t *psvr;
+
+	for (psvr = GET_NEXT(peersvrl);
+	     psvr; psvr = GET_NEXT(psvr->mi_link)) {
+		connect_to_peersvr(psvr);
```

Comment:
Don't we need to collect the return value of connect_to_peersvr() ?

Reply:
The error is logged at the inner level functions.